sdk/python/lib/pulumi/resource.py

```diff
@@ -49,6 +53,160 @@ def __init__(self,
         self.delete = delete
 
 
+def inherited_child_alias(
+        child_name: str,
+        parent_name: str,
+        parent_alias: 'Input[str]',
+        child_type: str) -> 'Output[str]':
+    """
+    inherited_child_alias computes the alias that should be applied to a child based on an alias
+    applied to it's parent. This may involve changing the name of the resource in cases where the
+    resource has a named derived from the name of the parent, and the parent name changed.
+    """
+
+#   If the child name has the parent name as a prefix, then we make the assumption that it was
+#   constructed from the convention of using `{name}-details` as the name of the child resource.  To
+#   ensure this is aliased correctly, we must then also replace the parent aliases name in the prefix of
+#   the child resource name.
+#
+#   For example:
+#   * name: "newapp-function"
+#   * opts.parent.__name: "newapp"
+#   * parentAlias: "urn:pulumi:stackname::projectname::awsx:ec2:Vpc::app"
+#   * parentAliasName: "app"
+#   * aliasName: "app-function"
+#   * childAlias: "urn:pulumi:stackname::projectname::aws:s3/bucket:Bucket::app-function"
+    from .output import Output as Op
```

Comment:
nope!  because ten during the type-checking phase, they collide.  and if i try to unilaterally import Output at the top of hte file, we get circularities in imports.